### ISSUE TITLE
Calcite dropdown item with links not being read correctly by the screen reader

### ISSUE THREAD

**Issue by Vani8021** [Esri/calcite-components#6663]
### Summary

When using screen reader to read the context in the dropdown items, it says `group` where it is expected to say as a `link`. This happens to the first item in the dropdown ONLY. 

Screenshot of code 
![image](https://user-images.githubusercontent.com/10887426/227664847-5ec52362-4b00-4070-bff7-c08508aa5806.png)

### Actual Behavior

Screenshots of what is being read by the screen reader

when focus in on `Esri` - This is NOT correct
![image](https://user-images.githubusercontent.com/10887426/227665469-dc257355-d81c-42df-b840-e161626fd443.png)

when focus in on `Google` - This is correct
![image](https://user-images.githubusercontent.com/10887426/227665573-3c0c99c5-bfea-4260-8f99-8db24244933d.png)

when focus is on `AGOL` - This is correct
![image](https://user-images.githubusercontent.com/10887426/227665592-be10de3f-1b5f-4765-9d79-c9a91e258409.png)

### Expected Behavior

When focus is on `Esri` - It should read it as `link` instead of `group`

### Reproduction Sample

https://codepen.io/vanisundar/pen/poOxaMN

### Reproduction Steps

Open screen reader and navigate thru the dropdown.

### Reproduction Version

1.0.7

### Priority impact

p1 - need for current milestone

### Esri team

ArcGIS Dashboards

**Comment by geospatialem** [comment]
@Vani8021 Unable to reproduce with some of the screen reader combinations including Windows with JAWS and NVDA or with VoiceOver on Mac with Safari.

Can you provide more details with the setup, including the Safari version and what OS is installed.

**Comment by Vani8021** [comment]
Mac OS - Chrome version -Version 111.0.5563.110

**Comment by geospatialem** [comment]
@Vani8021 Summarizing our discussions from chat prior to closing, where:

1. VoiceOver performed as-expected in the preferred Safari browser
2. NVDA and Firefox had some repetition as a result of NVDA's usage, which can be mitigated on top of Calcite for a better experience
